File: HotelBookingSystem/src/Shared/HotelBooking.Models/Entities/Customer.cs
using System;
using System.Collections.Generic;

namespace HotelBooking.Models.Entities
{
    public class Customer
    {
        public Guid CustomerId { get; set; }
        public string Email { get; set; } = string.Empty;
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string? PhoneNumber { get; set; }
        public string? Address { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        // Navigation properties
        public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();

        public Customer()
        {
            CustomerId = Guid.NewGuid();
            CreatedAt = DateTime.UtcNow;
            UpdatedAt = DateTime.UtcNow;
        }
    }
}

File: HotelBookingSystem/hotel-booking-frontend/src/types/customer.ts
export interface Customer {
  customerId: string;
  googleId?: string;
  email: string;
  firstName: string;
  lastName: string;
  phoneNumber?: string;
  address?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateCustomerDto {
  email: string;
  firstName: string;
  lastName: string;
  phoneNumber?: string;
  address?: string;
  googleId?: string;
}
